packages/core/src/components/slider/slider.tsx

```diff
@@ -31,72 +34,32 @@ export interface ISliderProps extends ICoreSliderProps {
     onRelease?(value: number): void;
 }
 
-export class Slider extends CoreSlider<ISliderProps> {
+export class Slider extends AbstractPureComponent<ISliderProps> {
     public static defaultProps: ISliderProps = {
-        disabled: false,
+        ...CoreSlider.defaultProps,
         initialValue: 0,
-        labelStepSize: 1,
-        max: 10,
-        min: 0,
-        showTrackFill: true,
-        stepSize: 1,
         value: 0,
-        vertical: false,
     };
 
     public static displayName: "Blueprint.Slider";
 
-    private handle: Handle;
-
-    protected renderFill() {
-        const { tickSizeRatio } = this.state;
-        const initialValue = clamp(this.props.initialValue, this.props.min, this.props.max);
-
-        let offsetRatio = (initialValue - this.props.min) * tickSizeRatio;
-        let sizeRatio = (this.props.value - initialValue) * tickSizeRatio;
-
-        if (sizeRatio < 0) {
-            offsetRatio += sizeRatio;
-            sizeRatio = Math.abs(sizeRatio);
-        }
-
-        const offsetPercentage = formatPercentage(offsetRatio);
-        const sizePercentage = formatPercentage(sizeRatio);
-
-        const style: React.CSSProperties = this.props.vertical
-            ? { bottom: offsetPercentage, height: sizePercentage }
-            : { left: offsetPercentage, width: sizePercentage };
-
-        return <div className={`${Classes.SLIDER}-progress`} style={style} />;
-    }
-
-    protected renderHandles() {
-        // make sure to *not* pass this.props.className to handle
+    public render() {
+        const { initialValue, value, onChange, onRelease, ...props } = this.props;
         return (
-            <Handle
-                {...this.props}
-                {...this.state}
-                className=""
-                label={this.formatLabel(this.props.value)}
-                ref={this.handleHandleRef}
-            />
+            <MultiSlider {...props} onChange={this.getHandler(onChange)} onRelease={this.getHandler(onRelease)}>
+                <SliderHandle
+                    value={value}
+                    trackIntentAfter={value < initialValue ? Intent.PRIMARY : undefined}
+                    trackIntentBefore={value > initialValue ? Intent.PRIMARY : undefined}
+                />
+                <SliderTrackStop value={initialValue} />
+            </MultiSlider>
         );
     }
 
-    protected handleTrackClick(event: React.MouseEvent<HTMLElement>) {
-        if (this.handle != null) {
-            this.handle.beginHandleMovement(event);
-        }
-    }
-
-    protected handleTrackTouch(event: React.TouchEvent<HTMLElement>) {
-        if (this.handle != null) {
-            this.handle.beginHandleTouchMovement(event);
-        }
+    private getHandler(callback: (value: number) => void) {
```

Comment:
not worth the render penalty here (new instances each invoke). just make two `handleChange` and `handleRelease` members.